mmv1/products/cloudquotas/QuotaPreference.yaml

```diff
@@ -0,0 +1,164 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'QuotaPreference'
+description: |
+  QuotaPreference represents the preferred quota configuration specified for a project, folder or organization. There is only one QuotaPreference resource for a quota value targeting a unique set of dimensions.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Cloud Quotas Overview': 'https://cloud.google.com/docs/quotas/overview'
+  api: 'https://cloud.google.com/docs/quotas/reference/rest/v1/projects.locations.quotaPreferences'
+base_url: '{{parent}}/locations/global/quotaPreferences'
+self_link: '{{parent}}/locations/global/quotaPreferences/{{quota_preference_id}}'
+create_url: '{{parent}}/locations/global/quotaPreferences?quotaPreferenceId={{quota_preference_id}}&ignoreSafetyChecks={{ignore_safety_checks}}'
+update_url: '{{parent}}/locations/global/quotaPreferences/{{quota_preference_id}}?allowMissing={{allow_missing}}&validateOnly={{validate_only}}&ignoreSafetyChecks={{ignore_safety_checks}}'
+update_verb: :PATCH
+update_mask: true
+skip_delete: true
+autogen_async: false
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/quota_preference.go.erb
+
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "cloudquotas_quota_preference_basic"
+    primary_resource_id: "preference"
+    skip_test: true
```

Comment:
Am I understanding correctly the this test is skipped because it is covered by the handwritten tests?

Reply:
This test is skipped because the resource can only be created once and cannot be deleted (our API has no deletion method). Recreating the same resource would cause API throwing an exception. The handwritten test is also failing due to the same reason, I'm wondering is there any workaround solutions?

Reply:
You may need to create a new project as part of the test, and then apply the quota to that project, in order to run this test nightly. But within a single run, I think it is fair to keep this example skipped (to reduce run time of tests, and the number of additional projects created).

There are many examples in the codebase you can see where a `google_project` is being created within the test config, but let me know if you need help.

Reply:
Added project creation as part of the test. Could you help run `/gcbrun`? It doesn't seem to respond when I trigger it.

Reply:
Ah yep, `/gcbrun` can only be trigger by our team at the moment.